### PR TITLE
Update versions and CHANGELOG for 9.1.0 release

### PR DESCRIPTION
**Pull request by alamb**
re #1304 

See rendered version https://github.com/alamb/arrow-rs/blob/alamb/prepare_for_10.0.0_release/CHANGELOG.md

There are some very nice performance improvements from @tustvold , some nice usability improvements from @viirya  and @liukun4515 , exciting parquet work from @shanisolomon / @zeevm , and code beautification @gsserge  

What a team 🚀 

cc @nevi-me

### CODE REVIEW

**Comment by tustvold** [comment]
Perhaps if there are no breaking changes this might be released as 9.1.0?

File: dev/release/README.md [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
 We try to release a new version of Arrow every two weeks. This cadence balances getting new features into arrow without overwhelming downstream projects with too frequent changes.
 
-If any code has been merged to master that has a breaking API change, as defined in [Rust RFC 1105](https://github.com/rust-lang/rfcs/blob/master/text/1105-api-evolution.md), the major version number incremented changed (e.g. `9.0.2` to `9.0.2`). Otherwise the new minor version incremented (e.g. `9.0.2` to `7.1.0`).
+If any code has been merged to master that has a breaking API change, as defined in [Rust RFC 1105](https://github.com/rust-lang/rfcs/blob/master/text/1105-api-evolution.md), the major version number incremented changed (e.g. `10.0.0` to `10.0.0`). Otherwise the new minor version incremented (e.g. `10.0.0` to `7.1.0`).
```
Comment (tustvold) [inline dev/release/README.md:26]: This line appears a bit broken :sweat_smile:

File: CHANGELOG.md [hunk 2]
```diff
@@ -19,6 +19,64 @@
...
+- Skip zero-ing primitive nulls [\#1280](https://github.com/apache/arrow-rs/pull/1280) [[parquet](https://github.com/apache/arrow-rs/labels/parquet)] ([tustvold](https://github.com/tustvold))
+- Add specialized filter kernels in `compute` module \(up to 10x faster\) [\#1248](https://github.com/apache/arrow-rs/pull/1248) [[parquet](https://github.com/apache/arrow-rs/labels/parquet)] [[arrow](https://github.com/apache/arrow-rs/labels/arrow)] ([tustvold](https://github.com/tustvold))
+
+**Closed issues:**
+
+- Expose column and offset index metadata offset [\#1317](https://github.com/apache/arrow-rs/issues/1317)
+- Expose bloom filter metadata offset [\#1308](https://github.com/apache/arrow-rs/issues/1308)
+- Improve ergonomics to construct `DictionaryArrays` from `Key` and `Value` arrays [\#1299](https://github.com/apache/arrow-rs/issues/1299)
+- Make it easier to iterate over `DictionaryArray` [\#1295](https://github.com/apache/arrow-rs/issues/1295) [[arrow](https://github.com/apache/arrow-rs/labels/arrow)]
+- Don't Interwine Bit and Byte Aligned Operations in `BitReader` [\#1282](https://github.com/apache/arrow-rs/issues/1282)
```
Comment (tustvold) [inline CHANGELOG.md:59]: This was actually just closed, without being fixe

File: dev/release/README.md (hunk 1 above)
Comment (alamb) [reply dev/release/README.md:26]: `sed` !

**Comment by alamb** [comment]
> Perhaps if there are no breaking changes this might be released as 9.1.0?

Good idea -- I will do so

File: CHANGELOG.md (hunk 2 above)
Comment (alamb) [reply CHANGELOG.md:59]: I think that is why it is listed under `Closed Issues` -- I'll make it explicit that it wasn't fixed

File: dev/release/README.md (hunk 1 above)
Comment (alamb) [reply dev/release/README.md]: fixed in 4b7a65e438

**Comment by alamb** [comment]
FWIW if we merge any other PRs we'll have to touch up this PR (rerun the https://github.com/apache/arrow-rs/blob/master/dev/release/update_change_log.sh as described in https://github.com/apache/arrow-rs/blob/master/dev/release/README.md#prepare-changelog-and-version

**Comment by alamb** [comment]
Here is a PR to DataFusion showing a clean upgrade to 9.1: https://github.com/apache/arrow-datafusion/pull/1851

**Review by nevi-me — APPROVED** [review]
Something that's a bit unclear @alamb is whether I should merge this before cutting an RC. It makes sense to do so, but not explicitly documented in the release docs (unless I missed it).

**Comment by alamb** [comment]
> Something that's a bit unclear @alamb is whether I should merge this before cutting an RC. It makes sense to do so, but not explicitly documented in the release docs (unless I missed it).

You are right -- this should be merged prior to cutting the RC -- thank you. I'll make a note to update the instructions when I am back if not already done so

**Comment by alamb** [comment]
PR to clarify instructions: https://github.com/apache/arrow-rs/pull/1370

**Pull request merged**